minor compaction never retries

I came across a situation where a subset of minor compactions failed and were never retried (version 1.9.3).  This then resulted in a major compaction hanging in the TabletMemory.waitForMinC method forever.  Eventually the system backed with files on those tablets.
The reason the minor compaction thread died was because our system was being modified with a new context and the VFSClassLoader was changing classpaths.  This resulted in a NoClassDefFoundError while trying to load the iterators for that table.  The purpose of this ticket is not to fix anything to do with the VFSClassLoader mechanism, but rather to ensure that if a minor compaction fails for any reason, it will be subsequently retried perhaps with some sort of exponential backoff.

The stack trace of the exception was something like this:
```
...
Caused by: java.lang.RuntimeException: Exception occurred during minor compaction on ...
   at org.apache.accumulo.tserver.tablet.Tablet.minorCompact
   at org.apache.accumulo.tserver.tablet.MinorCompactionTask.run
Caused by: java.lang.NoClassDefFoundError: Could not initialize class ...
   ...
   at org.apache.accumulo.core.iterators.IteratorUtil.loadIterators
   at org.apache.accumulo.tserver.tablet.Comparactor.compactLocalityGroup
   at org.apache.accumulo.tserver.tablet.Compactor.call
   at org.apache.accumulo.tserver.table.MinorCompactor.call
   at org.apache.accumulo.tserver.tablet.Tablet.minorCompact
```

--- a/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java
+++ b/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java
@@ -138,7 +138,7 @@ public class MinorCompactor extends Compactor {
           ProblemReports.getInstance(tabletServer).report(new ProblemReport(
               getExtent().getTableId(), ProblemType.FILE_WRITE, outputFileName, e));
           reportedProblem = true;
-        } catch (RuntimeException e) {
+        } catch (RuntimeException | NoClassDefFoundError e) {
           // if this is coming from a user iterator, it is possible that the user could change the
           // iterator config and that the
           // minor compaction would succeed

